### PR TITLE
EPMLABSBRN-917 BE: redesign providing/generation audio files

### PR DESCRIPTION
**Pull request by andrsam**
[EPMLABSBRN-917](https://jira.epam.com/jira/browse/EPMLABSBRN-917)

**Description**: add parameter for dynamically generate audio file url and handle it

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/service/TaskService.kt [hunk 1]
```diff
@@ -21,16 +22,22 @@ class TaskService(
...
         val tasks = taskRepository.findTasksByExerciseIdWithJoinedAnswers(exerciseId)
-        tasks.forEach { task ->
-            task.answerOptions
-                .forEach { resource -> wordsService.getFullS3UrlForWord(resource.word, resource.locale) }
+        if (!isAudioFileUrlGenerated) {
+            tasks.forEach { task -> processAnswerOptions(task) }
         }
         return when (val type = ExerciseType.valueOf(exercise.subGroup!!.series.type)) {
-            ExerciseType.SINGLE_SIMPLE_WORDS, ExerciseType.FREQUENCY_WORDS -> tasks.map { task -> task.toWordsSeriesTaskDto(type) }
+            ExerciseType.SINGLE_SIMPLE_WORDS, ExerciseType.FREQUENCY_WORDS -> tasks.map { task ->
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/service/TaskService.kt:36]: let's avoid such format changes in commits to not make git history dirty

File: src/main/kotlin/com/epam/brn/service/TaskService.kt [hunk 2]
```diff
@@ -42,10 +49,11 @@ class TaskService(
         log.debug("Searching task with id=$taskId")
         val task =
             taskRepository.findById(taskId).orElseThrow { EntityNotFoundException("No task found for id=$taskId") }
-        task.answerOptions.forEach { resource ->
-            resource.audioFileUrl =
-                wordsService.getFullS3UrlForWord(resource.word, resource.locale)
+
+        if (!isAudioFileUrlGenerated) {
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/service/TaskService.kt:53]: we don not use {} in one line code condition. )

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -69,13 +69,13 @@ yandex.speeds=0.8,1,1.2
 yandex.folderForFiles=frontend/public
 withMp3Conversion=false
 withSavingToS3=true
-
 brn.audio.file.default.path=default/%s.mp3
 brn.audio.file.second.series.path=series2/%s.mp3
+brn.audio.file.url.generate.dynamically=false
+brn.audio.file.series.path=series%d/%s.mp3
```
Comment (ElenaSpb) [inline src/main/resources/application.properties:75]: do we really need this prop?

**Review by lifeart — APPROVED** [review]

**Review by lifeart — APPROVED** [review]

File: src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt [hunk 4]
```diff
@@ -15,5 +15,6 @@ data class ExerciseDto(
     var signals: MutableList<SignalTaskDto> = mutableListOf(),
     var active: Boolean = true,
     var changedBy: String? = "InitialDataLoader",
-    var changedWhen: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC)
+    var changedWhen: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC),
+    var isAudioFileUrlGenerated: Boolean = false
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt:19]: could you please put this changes and into Exercise json response too

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**